kin-sdk-core/src/androidTest/java/kin/sdk/core/RequestTest.java

```diff
@@ -0,0 +1,196 @@
+        CountDownLatch latch = new CountDownLatch(1);
+        AtomicInteger integer = new AtomicInteger(0);
+        Request<Object> dummyRequest = new Request<>(() -> {
+            integer.incrementAndGet();
+            return new Object();
+        });
+        dummyRequest.run(new ResultCallback<Object>() {
+            @Override
+            public void onResult(Object result) {
+                integer.incrementAndGet();
+            }
+
+            @Override
+            public void onError(Exception e) {
+                integer.incrementAndGet();
+            }
+        });
+        dummyRequest.cancel();
+        latch.await(TIMEOUT_DURATION_MILLIS, TimeUnit.MILLISECONDS);
+        assertEquals(0, integer.get());
+    }
+
+    @Test
+    public void run_cancelAfterRun() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(1);
+        AtomicBoolean threadInterrupted = new AtomicBoolean(false);
+        Request<Object> dummyRequest = new Request<>(() -> {
+            try {
+                Thread.sleep(TASK_DURATION_MILLIS);
+            } catch (InterruptedException ie) {
+                threadInterrupted.set(true);
+                latch.countDown();
+            }
+            return new Object();
+        });
+        dummyRequest.run(new ResultCallback<Object>() {
+            @Override
+            public void onResult(Object result) {
+            }
+
+            @Override
+            public void onError(Exception e) {
+            }
+        });
+        Thread.sleep(TASK_DURATION_MILLIS / 2);
+        dummyRequest.cancel();
+        assertTrue(latch.await(TIMEOUT_DURATION_MILLIS, TimeUnit.MILLISECONDS));
+        assertEquals(true, threadInterrupted.get());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void request_nullCallable() {
+        new Request<>(null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void run_nullCallback() {
+        Request<String> request = new Request<>(() -> "");
+        request.run(null);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void run_twice() {
+        Request<String> request = new Request<>(() -> "");
+        request.run(new ResultCallback<String>() {
+            @Override
+            public void onResult(String result) {
+            }
+
+            @Override
+            public void onError(Exception e) {
+            }
+        });
+        request.run(new ResultCallback<String>() {
+            @Override
+            public void onResult(String result) {
+
+            }
+
+            @Override
+            public void onError(Exception e) {
+
+            }
+        });
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void runAfterCancel() {
+        Request<String> request = new Request<>(() -> "");
+        request.cancel();
+        request.run(new ResultCallback<String>() {
```

Comment:
maybe you can create a method: getEmptyResultCallback(),
i see you are using it in few places

Reply:
:+1:
(`getEmptyRequest()` it's more complicated as in one case there's a `cancel` in between.)